examples/hal_cpu_remote_server/README.md

```diff
@@ -0,0 +1,127 @@
+# Hal server
+
+## Introduction
+
+This supports a remote CPU based HAL which uses a socket connection to
+communicate with a remote client. This can be cross-compiled as needed. It is a
+small program which can be modified to use different ways of
+transmission or different underlying HAL interfaces.
+
+It should be paired with a built `oneapi-construction-kit` OpenCL interface using
+the HAL socket client, which resides under `examples/hal_cpu_client`. Details of
+how to build this target will be given below.
+
+## Security
+
+We recommend not running this program as root to reduce any likelihood of a
+malicious actor, as this gives a protocol for running programs. The customer
+should consider a specific user account just for running the server that has
+limited access.
+
+We also recommend using ssh port forwarding. Note that we do not add any
+additional encryption of the executables that are sent.
+
+## Building the server
+
+This can be done as follows:
+
+```
+  cmake -GNinja <path_to_ock>/examples/hal_cpu_remote_server -Bbuild_server
+  ninja
+```
+
+If cross-compilation (for example RISC-V) is needed the following additional arguments are needed:
+
+```
+   -DCMAKE_TOOLCHAIN_FILE=<path_to_ock>/platform/riscv64-linux/riscv64-gcc-toolchain.cmake -DCMAKE_CROSSCOMPILING=TRUE
+```
+
+Note this requires the RISC-V toolchain to be installed as follows on Ubuntu:
+
+```
+  sudo apt-get install --yes gcc-9-riscv64-linux-gnu g++-9-riscv64-linux-gnu
+```
+
+If you wish to run on a non RISC-V machine, then `qemu` and the RISC-V toolchain will need to be installed.
+
+## Building the client
+
+We are looking to simplify this process, but currently we need to create an out
+of tree target. This is done as follows (for RISC-V):
```

Comment:
I could be misremembering, but didn't you want to change this process before merging?

Reply:
I think it's a big enough change, we should consider doing it separately and not hold up this PR.